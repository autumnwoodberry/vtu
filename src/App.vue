<template>
  <div id="app">
    <div>
      <pre>{{ $data }}</pre>
      <p>The "a" boolean toggles the wrapper div on/off with a "fade" transition.</p>
      <p>The "b" boolean controls which child div is shown with a "slide" transition.<p>
      <div>
        <input type="checkbox" v-model="a">
        <label>a</label>
      </div>
      <div>
        <input type="checkbox" v-model="b">
        <label>b</label>
      </div>
    </div>
    <transition name="fade">
      <div v-if="a" style="position: relative; width: 100px; height: 100px; overflow: hidden; border: 1px solid black;">
        <transition name="slide-left">
          <div v-if="!b" style="background-color: lightblue; position: absolute; width: 100px; height: 100px;">
            <pre>a && !b</pre>
          </div>
        </transition>
        <transition name="slide-right">
          <div v-if="b" style="background-color: lightgreen; position: absolute; width: 100px; height: 100px;">
            <pre>a && b</pre>
          </div>
        </transition>
      </div>
    </transition>
  </div>
</template>

<script>
export default {
  name: 'vtu',
  data () {
    return {
      a: false,
      b: false
    }
  }
}
</script>

<style>
  /* slide left */
  .slide-left-enter-active, .slide-left-leave-active { transition: all 500ms ease; }
  .slide-left-enter { transform: translateX(-100%); }
  .slide-left-enter-to { transform: translateX(0); }
  .slide-left-leave-to { transform: translateX(-100%); }

  /* slide right */
  .slide-right-enter-active, .slide-right-leave-active { transition: all 500ms ease; }
  .slide-right-enter { transform: translateX(100%); }
  .slide-right-enter-to { transform: translateX(0); }
  .slide-right-leave-to { transform: translateX(100%); }

  /* fade */
  .fade-enter-active, .fade-leave-active { transition: all 500ms ease; }
  .fade-enter, .fade-leave-active { opacity: 0; }
</style>
