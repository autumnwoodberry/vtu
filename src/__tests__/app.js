import Vue from 'vue'
import { mount } from 'vue-test-utils'
import App from '../App'

let wrapper

beforeEach(() => {
    wrapper = mount(App)
})

// without nexttick (all snapshots are identical)

test('!a && !b', () => {
    wrapper.vm.a = false
    wrapper.vm.b = false

    expect(wrapper.vm.a).toBe(false)
    expect(wrapper.vm.b).toBe(false)
    expect(wrapper.html()).toMatchSnapshot()
})

test('a && !b', () => {
    wrapper.vm.a = true
    wrapper.vm.b = false

    expect(wrapper.vm.a).toBe(true)
    expect(wrapper.vm.b).toBe(false)
    expect(wrapper.html()).toMatchSnapshot()
})

test('!a && b', () => {
    wrapper.vm.a = false
    wrapper.vm.b = true

    expect(wrapper.vm.a).toBe(false)
    expect(wrapper.vm.b).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
})

// with nexttick (snapshots show correct html as expected)

test('!a && !b using Vue.nextTick', () => {
    wrapper.vm.a = false
    wrapper.vm.b = false

    expect(wrapper.vm.a).toBe(false)
    expect(wrapper.vm.b).toBe(false)
    Vue.nextTick(() => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})

test('a && !b using Vue.nextTick', () => {
    wrapper.vm.a = true
    wrapper.vm.b = false

    expect(wrapper.vm.a).toBe(true)
    expect(wrapper.vm.b).toBe(false)
    Vue.nextTick(() => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})

test('!a && b using Vue.nextTick', () => {
    wrapper.vm.a = false
    wrapper.vm.b = true

    expect(wrapper.vm.a).toBe(false)
    expect(wrapper.vm.b).toBe(true)
    Vue.nextTick(() => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})
