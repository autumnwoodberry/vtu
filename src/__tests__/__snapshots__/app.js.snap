// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`!a && !b 1`] = `
<div id="app">
    <div>
        <pre>{
  "a": false,
  "b": false
}</pre>
        <p>The "a" boolean toggles the wrapper div on/off with a "fade" transition.</p>
        <p>The "b" boolean controls which child div is shown with a "slide" transition.</p>
        <p></p>
        <div><input type="checkbox"> <label>a</label></div>
        <div><input type="checkbox"> <label>b</label></div>
    </div>
    <!---->
</div>
`;

exports[`!a && !b using Vue.nextTick 1`] = `
<div id="app">
    <div>
        <pre>{
  "a": false,
  "b": false
}</pre>
        <p>The "a" boolean toggles the wrapper div on/off with a "fade" transition.</p>
        <p>The "b" boolean controls which child div is shown with a "slide" transition.</p>
        <p></p>
        <div><input type="checkbox"> <label>a</label></div>
        <div><input type="checkbox"> <label>b</label></div>
    </div>
    <!---->
</div>
`;

exports[`!a && b 1`] = `
<div id="app">
    <div>
        <pre>{
  "a": false,
  "b": false
}</pre>
        <p>The "a" boolean toggles the wrapper div on/off with a "fade" transition.</p>
        <p>The "b" boolean controls which child div is shown with a "slide" transition.</p>
        <p></p>
        <div><input type="checkbox"> <label>a</label></div>
        <div><input type="checkbox"> <label>b</label></div>
    </div>
    <!---->
</div>
`;

exports[`!a && b using Vue.nextTick 1`] = `
<div id="app">
    <div>
        <pre>{
  "a": false,
  "b": true
}</pre>
        <p>The "a" boolean toggles the wrapper div on/off with a "fade" transition.</p>
        <p>The "b" boolean controls which child div is shown with a "slide" transition.</p>
        <p></p>
        <div><input type="checkbox"> <label>a</label></div>
        <div><input type="checkbox"> <label>b</label></div>
    </div>
    <!---->
</div>
`;

exports[`a && !b 1`] = `
<div id="app">
    <div>
        <pre>{
  "a": false,
  "b": false
}</pre>
        <p>The "a" boolean toggles the wrapper div on/off with a "fade" transition.</p>
        <p>The "b" boolean controls which child div is shown with a "slide" transition.</p>
        <p></p>
        <div><input type="checkbox"> <label>a</label></div>
        <div><input type="checkbox"> <label>b</label></div>
    </div>
    <!---->
</div>
`;

exports[`a && !b using Vue.nextTick 1`] = `
<div id="app">
    <div>
        <pre>{
  "a": true,
  "b": false
}</pre>
        <p>The "a" boolean toggles the wrapper div on/off with a "fade" transition.</p>
        <p>The "b" boolean controls which child div is shown with a "slide" transition.</p>
        <p></p>
        <div><input type="checkbox"> <label>a</label></div>
        <div><input type="checkbox"> <label>b</label></div>
    </div>
    <div style="position: relative; width: 100px; height: 100px; overflow: hidden; border: 1px solid black;" class="fade-enter fade-enter-active">
        <div style="position: absolute; width: 100px; height: 100px;">
            <pre>a &amp;&amp; !b</pre>
        </div>
        <!---->
    </div>
</div>
`;
